sigopt/hyperopt/base.py

```diff
@@ -0,0 +1,223 @@
+from .compat import Trials, STATUS_OK, STATUS_FAIL, fmin
+from .. import SigOptFactory
+
+HYPEROPT_SOURCE_NAME = 'Hyperopt Suggest'
+HYPEROPT_SOURCE_PRIORITY = 1
+
+class SigOptTrials(Trials):
+  def __init__(self, project, trials=None, online=True):
+    self.factory = SigOptFactory(project)
+    self.online = online
+    self._trials = trials if trials else Trials()
+    self.uploaded_tids = {}
+
+  @property
+  def _dynamic_trials(self):
+    return self._trials._dynamic_trials
+
+
+  @property
+  def parameters(self):
+    return [self.trial_parameters(trial) for trial in self.trials]
+
+  def upload(self, trials=None):
+    new_trials = []
+    trials = trials if trials is not None else self.trials
+    for trial in trials:
+      result = trial['result']
+      status = result.get('status')
+      if status in [STATUS_OK, STATUS_FAIL]:
+        tid = trial['tid']
+        if tid not in self.uploaded_tids:
+          new_trials.append(trial)
+    ids = self._upload(new_trials)
+    self.uploaded_tids.update(ids)
+    return ids
+
+
+  def _upload(self, trials):
```

Comment:
I was expecting this to be `_upload_trial` of only one trial. Not sure why you are filtering them and then adding them. 

But in any case, we need to make sure that we can create so many runs that fast. For the offline case, this could be problematic, so we might need to split the list somehow or add a delay.

We probably should think about this in the context of the python client's `factory.create_multiple_runs(metadata=metadata, list_of_runs_attributes)` 
where `list_of_runs_attributes` is a list of dicts.
What do you think about that?

Reply:
I think batch upload API is more efficient and can be used for implement `_upload_trial`.